makiflow/models/segmentation/compile.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2020  Igor Kilbas, Danil Gribanov, Artem Mukhin
+#
+# This file is part of MakiFlow.
+#
+# MakiFlow is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# MakiFlow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Foobar.  If not, see <https://www.gnu.org/licenses/>.
+
+from .training_modules import FocalTrainingModule, MakiTrainingModule, QuadraticCrossEntropyTrainingModule, \
+    WeightedFocalTrainingModule, WeightedCrossEntropyTrainingModule
+
+import json
+from makiflow.base.maki_entities import MakiCore
+from .main_modules import SParams
+
+
+class Segmentator(
+    FocalTrainingModule,
+    MakiTrainingModule,
+    QuadraticCrossEntropyTrainingModule,
+    WeightedCrossEntropyTrainingModule,
+    WeightedFocalTrainingModule
+):
+
+    @staticmethod
+    def from_json(path_to_model, batch_size=None, generator=None):
```

Comment:
It is better to change the `generator` argument to an arbitrary `input_tensor`. It is not necessary that the input is generator, it can be an arbitrary MakiTensor.